src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -507,5 +507,26 @@ public static int countEmptyStrings(String... strings) {
      */
     public static String convertToEmptyStringIfNull(String str) {
         return (str == null) ? "" : str;
-    }  
+    }
+    
+    /**
+     * Removes the outermost enclosing square brackets surrounding a string.
+     * 
+     * @param str
+     * @return the string without the outermost enclosing square brackets and 
+     *         without leading and trailing spaces, if the given string is enclosed by square brackets <br/> 
+     *         the string itself if the given string is not enclosed by square brackets <br/>
+     *         null if the given string is null
+     */
+    public static String removeEnclosingSquareBrackets(String str) {
+        if (str == null) {
+            return null;
+        }
+        
+        String trimmedString = str.trim();
+        if (!trimmedString.startsWith("[") || !trimmedString.endsWith("]")) {
+            return str;
+        }
+        return trimmedString.substring(1, trimmedString.length() - 1).trim();
```

Comment:
This line is the main logic of your method. Let's separate it from the rest using a blank line.